moveit_core/collision_detection/include/moveit/collision_detection/collision_common.h

```diff
@@ -161,6 +167,31 @@ struct CollisionResult
     cost_sources.clear();
   }
 
+  /** \brief Throttled log of the first collision pair, if any */
+  void logFirstCollisionPair()
+  {
+    if (!contacts.empty())
+    {
+      ROS_WARN_STREAM_THROTTLE_NAMED(ROS_LOG_THROTTLE_PERIOD, LOGNAME,
+                                     "Objects in collision (among others, possibly): "
```

Comment:
```suggestion
                                     "Objects in collision (printing first collision pair of " << contacts.size() << "): "
```

Reply:
thanks! Will make this change.

You're right, the previous functions printed the same thing but that was only in moveit_servo. The change here would allow somebody (for example) to print the links that are in collision while working through this tutorial:

https://ros-planning.github.io/moveit_tutorials/doc/planning_scene/planning_scene_tutorial.html

They just need to add `collision_request.logFirstCollisionPair();`

Reply:
Oh, I didn't see that the deleted function was in `moveit_servo` and not `collision_common`. A printout like that would be useful, I agree. 

I don't have a strong opinion, but I also liked that DEBUG printed all collision pairs by default. I would keep it combined and call the function `print()`. Maybe @v4hn has a comment.

Also, if `CollisionResult` gets its own member function it should be a class and not a struct ([Google style guidelines](https://google.github.io/styleguide/cppguide.html#Structs_vs._Classes)).

Reply:
I have to push back on making it a class.

1) there already was a helper function there, clear()
2) the google style guide you linked says this about structs:  `Constructors, destructors, and helper methods may be present; however, these methods must not require or enforce any invariants.`
3) if it was a class, I would have to rename the members, and that would be a huge change

Reply:
I guess a `print()` function classifies as a helper function. I'm fine with this, just relaying it.